### PR TITLE
[FrontEnd] Use SetVector (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**
We could also use range-based for loops at several places, but I'm
leaving that to a subsequent patch.

### CODE REVIEW

File: clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp [hunk 1]
```diff
@@ -4082,8 +4080,10 @@ std::string RewriteModernObjC::SynthesizeBlockFunc(BlockExpr *CE, int i,
 
   // Create local declarations to avoid rewriting all closure decl ref exprs.
   // First, emit a declaration for all "by ref" decls.
-  for (SmallVectorImpl<ValueDecl *>::iterator I = BlockByRefDecls.begin(),
-       E = BlockByRefDecls.end(); I != E; ++I) {
+  for (llvm::SmallSetVector<ValueDecl *, 8>::iterator
```
Comment (nikic) [inline clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp:4083]: Use auto for iterators please.

File: clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp [hunk 2]
```diff
@@ -5157,14 +5163,12 @@ void RewriteModernObjC::CollectBlockDeclRefInfo(BlockExpr *Exp) {
     // Unique all "by copy" declarations.
     for (unsigned i = 0; i < BlockDeclRefs.size(); i++)
       if (!BlockDeclRefs[i]->getDecl()->hasAttr<BlocksAttr>()) {
```
Comment (nikic) [inline clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp:5165]: Can drop the braces now.

**Review by nikic — APPROVED** [review]

File: clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp (hunk 1 above)
Comment (kazutakahirata) [reply clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp]: Fixed in the latest iteration.  Thanks!

File: clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp (hunk 2 above)
Comment (kazutakahirata) [reply clang/lib/Frontend/Rewrite/RewriteModernObjC.cpp]: Fixed in the latest iteration.  Thanks!

**Pull request merged**